producer/src/main/kotlin/streams/kafka/KafkaEventRouter.kt

```diff
@@ -57,34 +59,40 @@ class KafkaEventRouter: StreamsEventRouter {
         StreamsUtils.ignoreExceptions({ kafkaAdminService.stop() }, Exception::class.java)
     }
 
-    private fun send(producerRecord: ProducerRecord<String, ByteArray>) {
+    private fun send(producerRecord: ProducerRecord<String, ByteArray>, sync: Boolean = false): Map<String, Any>? {
         if (!kafkaAdminService.isValidTopic(producerRecord.topic())) {
             if (log.isDebugEnabled) {
                 log.debug("Error while sending record to ${producerRecord.topic()}, because it doesn't exists")
             }
             // TODO add logging system here
-            return
+            return null
         }
-        producer.send(producerRecord) { meta, error ->
-            if (meta != null && log.isDebugEnabled) {
-                log.debug("Successfully sent record in partition ${meta.partition()} offset ${meta.offset()} data ${meta.topic()} key size ${meta.serializedKeySize()}")
-            }
-            if (error != null) {
-                if (log.isDebugEnabled) {
-                    log.debug("Error while sending record to ${producerRecord.topic()}, because of the following exception:", error)
+        return if (sync) {
+            producer.send(producerRecord).get().toMap()
+        } else {
+            producer.send(producerRecord) { meta, error ->
+                if (meta != null && log.isDebugEnabled) {
+                    log.debug("Successfully sent record in partition ${meta?.partition()} offset ${meta?.offset()} data ${meta?.topic()} key size ${meta?.serializedKeySize()}")
+                }
+                if (error != null) {
+                    if (log.isDebugEnabled) {
+                        log.debug("Error while sending record to ${producerRecord.topic()}, because of the following exception:", error)
+                    }
+                    // TODO add logging system here
                 }
             }
+            null
```

Comment:
This is a minor point, that does not require change, but sometimes javadocs or the kotlin equivalent is nice in this condition.  The function looks like it returns Map<String,Any> which you would think is the result of publishing.  But it doesn't really -- the return type of the function is only relevant if sync: true.  The caller must know when to "take the return value seriously", which in our case the caller does, so it's ok